pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -435,3 +437,7 @@ func hasInternalIP(machine *machinev1beta1.Machine, memberInternalIP string) boo
 	}
 	return false
 }
+
+func minimumTolerableQuorum(desiredControlPlaneReplicasCount int) int {
```

Comment:
we have something similar already in:
https://github.com/openshift/cluster-etcd-operator/blob/2ec6e4d7b2d573f99bdce632acbc46e01c855e0d/pkg/etcdcli/health.go#L241-L268

Can we reuse that instead?

Reply:
Yes I tried, but the conditions used here are too strict and this was the problem of scaling down 

I use `totalMembers < desiredControlPlaneCount` as a first check to decide if we can attempt to scale down, or skip.
```
if len(liveVotingMembers) < desiredControlPlaneReplicasCount {
		klog.V(2).Infof("Ignoring scale-down since the number of etcd members (%d) < desired number of control-plane replicas (%d) ", len(liveVotingMembers), desiredControlPlaneReplicasCount)
		return nil
}
```


I use  `len(healthyLiveVotingMembers) < minimumTolerableQuorum(desiredControlPlaneReplicasCount)` to decide if we have a chance to scale down without losing Quorum 

The conditions used in `IsQuorumFaultTolerant()` is too strict for scaling down

Reply:
that's fair, but can we consolidate the formula for computing this into a shared func instead? so not everybody has to re-invent this math and test it again, with what parameters you feed the formula is up to the caller.

Reply:
I am fine with that, but both methods have different args  🤔

Reply:
@Elbehery You don't have to reuse the `IsQuorumFaultTolerant` function as it is or change its args but you can consolidate the quorum computation to a new function and reuse that in both `IsQuorumFaultTolerant` and in `minimumTolerableQuorum` so we're not maintaining the same thing in different packages.

Reply:
@hasbro17 @tjungblu  I tried refactoring  `minimumTolerableQuorum`  to `common` but it resulted in cyclic dependency .. I added it to etcdcli as exported